pypowerwall/cloud.py

```diff
@@ -906,8 +906,11 @@ def setup(self, email=None):
             else:
                 sitelabel = " "
             siteids.append(s["energy_site_id"])
-            print(" %s%d - %s (%s) - Type: %s" % (sitelabel, idx, s["site_name"], 
+            try:
+                print(" %s%d - %s (%s) - Type: %s" % (sitelabel, idx, s["site_name"],
                     s["energy_site_id"], s["resource_type"]))
+            except KeyError:
+                pass
```

Comment:
Thanks for this @dcgibbons !

Can you check to see if this would work instead of the try/except block?

```python
            if "site_name" in s and "resource_type" in s:
                print(" %s%d - %s (%s) - Type: %s" % (sitelabel, idx, s["site_name"],
                    s["energy_site_id"], s["resource_type"]))
            else:
                print(" %s%d - %s (%s) - Type: %s" % (sitelabel, idx, "Unknown",
                    s["energy_site_id"], "Unknown"))
```

That way it prints a placeholder for this bad site location and also keeps a consistent index for the user to select (versus skipping numbers).

Reply:
Yup! That works and I updated this PR with that.

```
2 Sites Found (* = default)
------------------------------------------------------------
 *1 - 200 Amp Panel (XXXXXX) - Type: battery
  2 - Unknown (YYYYYY) - Type: Unknown

  Select a site [1]: 1
```